src/vat/webkey.js

```diff
@@ -28,11 +29,8 @@ function canPassByCopy(val) {
     return false;
   }
   const names = Object.getOwnPropertyNames(val);
-  for (const name of names) {
-    if (typeof val[name] === 'function') {
-      return false;
-    }
-  }
+  const hasFunction = names.every(name => typeof val[name] === 'function');
```

Comment:
These aren't equivalent (if there's a mix of true and false values, the old one would `return false`, but the new one continues onwards), but I think replacing `every` with `some` should fix it.

```js
function a(val, names) {
  for (const name of names) {
    if (typeof val[name] === 'function') {
      return false;
    }
  }
  return true;
}

function b(val, names) {
  const hasFunction = names.every(name => typeof val[name] === 'function');
  if (hasFunction) return false;
  return true;
}

function c(val, names) {
  const hasFunction = names.some(name => typeof val[name] === 'function');
  if (hasFunction) return false;
  return true;
}

function both(val, names) {
  console.log(a(val, names), b(val, names), c(val, names), names);
}

const val = {
  f1() {},
  f2() {},
  nf1: 1,
  nf2: 2,
};

both(val, ['f1', 'f2']);
both(val, ['f1', 'nf1']);
both(val, ['nf1', 'nf2']);
```

```
$ node t.js
false false false [ 'f1', 'f2' ]
false true false [ 'f1', 'nf1' ]
true true true [ 'nf1', 'nf2' ]
```

Reply:
You're absolutely right.